.github/workflows/automatic-cherry-pick.yaml

```diff
@@ -0,0 +1,246 @@
+name: cherry-pick
+
+on:
+  issue_comment:
+    types: [created]
+
+jobs:
+  process_comment:
+    # This job will parse the cherry-pick command and perform several sanity checks.
+    # Ideally we want all pull requests opened from a given command to be correct and successfull, so that
+    # developers don't have to reissue a cherry-pick command for a subset of the commits/branches.
+    # Therefore, we do all possible sanity checks in this job, before any pull request is opened in subsequent jobs.
+    if: github.event.issue.pull_request != '' && contains(github.event.comment.body, 'cherry-pick')
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+    outputs:
+      commits: ${{ steps.command.outputs.commits }}
+      branches: ${{ steps.command.outputs.branches }}
+      branch_matrix: ${{ steps.command.outputs.branch_matrix }}
+      status: ${{ steps.report.outputs.status }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          # We need the history of all branches for the sanity checks
+          fetch-depth: 0
+
```

Comment:
It would be a good idea to check this was actually a pull request, and not just an issue.

In which case you could cease processing and ignore.

https://stackoverflow.com/a/59780123

Reply:
But you'd also need to set a value for the status so the subsequent workflows don't run
```
echo "status=failure" >> $GITHUB_OUTPUT
```

Reply:
> It would be a good idea to check this was actually a pull request, and not just an issue.

This is handled above with the following condition:

`if: github.event.issue.pull_request != ''`

Reply:
Sorry I missed that. Sweet.

Reply:
Feel free to resolve this. I don't have permission to do so.